packages/geoprocessing/src/toolbox/overlapFeatures.ts

```diff
@@ -56,11 +56,33 @@ export async function overlapFeatures(
   const { includeChildMetrics } = newOptions;
   let sumValue: number = 0;
   let isOverlap = false;
+  let featureIndices: Set<number> = new Set();
   const sketches = (Array.isArray(sketch) ? sketch : toSketchArray(sketch)).map(
     (s) => truncate(s)
   );
   const finalFeatures = features.map((f) => truncate(f));
 
+  // Create individual sketch metrics
+  const sketchMetrics: Metric[] = sketches.map((curSketch) => {
```

Comment:
For example, could you add the second `index` parameter that map provides here and use it to track the indices?  Same as below, just a higher level